src/web_resources/HangarXPLOR.Download.js

```diff
@@ -28,57 +28,111 @@ HangarXPLOR._callbacks = HangarXPLOR._callbacks || {};
+          item.package = 0; // Don't report package for remaining ships in this Package/Combo
+          ship.type = 'Ship';
+          ship.manufacturer = $('.liner span', $ship).text();
+          ship.manufacturer = _manufacturerShortMap[ship.manufacturer] || ship.manufacturer;
+          ship.name = $('.title', $ship).text();
+          ship.name = HangarXPLOR.CleanShipName(ship.name);
+          ship.name = ship.name.replace(/^\s*(?:Aegis|Anvil|Banu|Drake|Esperia|Kruger|MISC|Origin|RSI|Tumbril|Vanduul|Xi'an)[^a-z0-9]+/gi, '');
+          return ship;
+        })
+        // Sort any inner ships by manufacturer and then name
+        .sort(function(a, b) {
+          if (a.manufacturer < b.manufacturer) {
+            return -1
+          }
+          if (a.manufacturer > b.manufacturer) {
+            return 1
+          }
+          if (a.name < b.name) {
+            return -1
+          }
+          if (a.name > b.name) {
+            return 1
+          }
+          return 0;
+        })
+        .get()
+      }
+    })
+    .get();
   }
   
   HangarXPLOR._callbacks.DownloadJSON = function(e) {
     e.preventDefault();
     
-    var $target = $(HangarXPLOR._selected.length > 0 ? HangarXPLOR._selected : HangarXPLOR._inventory);
+    var items = HangarXPLOR.GetDownloadItems();
+
+    var buffer = JSON.stringify(items, null, 2);
+
+    // console.log('DownloadJSON buffer\n', buffer);
+    // return;
     
-    $download.attr('href', 'data:text/json;charset=utf-8,' + encodeURIComponent(JSON.stringify(HangarXPLOR.GetShipList($target), null, 2)));
-    $download.attr('download', 'shiplist.json');
+    $download.attr('href', 'data:text/json;charset=utf-8,' + encodeURIComponent(buffer));
+    $download.attr('download', 'HangarItems.json');
     $download.attr('type', 'text/json');
     $download[0].click();
   }
   
   HangarXPLOR._callbacks.DownloadCSV = function(e) {
     e.preventDefault();
     
-    var $target = $(HangarXPLOR._selected.length > 0 ? HangarXPLOR._selected : HangarXPLOR._inventory);
+    var items = HangarXPLOR.GetDownloadItems();
     
-    var buffer = "Manufacturer, Ship, Lti, Warbond, ID, Pledge, Cost, Date\n";
-    buffer = buffer + HangarXPLOR.GetShipList($target).map(function(ship) { return [ '"' + ship.manufacturer + '"', '"' + ship.name + '"', ship.lti, ship.warbond, ship.package_id, '"' + ship.pledge + '"', '"' + ship.cost + '"', '"' + ship.pledge_date + '"' ].join(',')}).join('\n')
+    var buffer = "Pledge,Type,Manufacturer,Name,ID,Cost,Date,LTI,Warbond,Package,Giftable\r\n";
+    buffer = buffer + items.map(function(item) {
+      return [
+        '"' + item.pledge + '"',
+        '"' + item.type + '"',
+        '"' + item.manufacturer + '"',
+        '"' + item.name + '"',
+        item.id,
+        item.cost,
+        '"' + item.date + '"',
+        item.lti,
+        item.warbond,
+        item.package,
+        item.giftable
+      ]
+      .join(',')
+    })
```

Comment:
Much easier to read

Reply:
Even nicer if I had used template literals